.github/workflows/pr-deploy.yaml

```diff
@@ -34,6 +34,17 @@ permissions:
   pull-requests: write # needed for commenting on PRs
 
 jobs:
+  check_membership:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if actor is a member
+        run: |
+          set -euo pipefail
+          response=$(curl -s -o /dev/null -w "%{http_code}" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/orgs/coder/members/${{ github.actor }}")
```

Comment:
Are you sure this works in each case? For example, <https://api.github.com/orgs/coder/members/matifali> returns
```json
{
  "message": "User does not exist or is not a public member of the organization",
  "documentation_url": "https://docs.github.com/rest/orgs/members#check-public-organization-membership-for-a-user"
}
```

Reply:
It seems like it works for private members:

```bash
$ curl -s -H "Authorization: token $(gh auth token)" "https://api.github.com/orgs/coder/members/matifali" -v 
...
< HTTP/2 204 
...
```

Reply:
I'm using https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user

Reply:
Yes, all works great. I was testing without the token and always getting a 302. Thank you for adding the check.

Reply:
It would be nice to test by deploying the current PR.

Reply:
I tried that in https://github.com/coder/coder/pull/13280 (where I originally had this fix) but GH didn't seem to use the new workflow. Is there a timeout maybe? I'll give it a try though.

Reply:
Damn, I tried to execute this but I got:

```bash
$ ./scripts/deploy-pr.sh -d -b -y                                                                           
branchName: dk/fix-preview
prNumber: 13404
experiments: 
build: true
deploy: true
could not create workflow dispatch event: HTTP 422: No ref found for: dk/fix-preview (https://api.github.com/repos/coder/coder/actions/workflows/60960476/dispatches)
```

I think the branch has to be in the main repo.
Strange that I didn't get this before, though.

Reply:
Oh, looks like I must've pushed `dk/verify-agent` (the branch I was testing with previously) to the [main repo](https://github.com/coder/coder/tree/dk/verify-agent) somehow?

Reply:
I am noticing the same issue. It's deploying the previous branch I was working on. `node-20`